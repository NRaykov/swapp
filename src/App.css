.App-header {
  width: 100%;
  margin-bottom: 2rem;
  background-color: #000000;
}
.logo {
  font-size: 3rem;
  font-family: 'StarWars';
}
.loginButton {
  width: 3.5rem;
  height: 3.5rem;
  border-radius: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #ffffff;
}
.loginButton svg {
  color: #72d5dc;
}

.loginButton:hover {
  background-color: #f8e35a;
}
.loginButton:hover svg {
  color: white;
}

