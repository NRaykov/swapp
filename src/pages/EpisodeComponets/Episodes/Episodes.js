import React from 'react';
import  Heading from "../../../components/StylesComponents/elements/heading/heading";
import Card from "../../../components/StylesComponents/elements/card/card"
import {CardImg, CardText, Row, Col} from 'reactstrap';
import styles from "./styles.module.css";
import { Link, Link as RouterLink } from 'react-router-dom';
import {useQuery} from "@apollo/react-hooks";
import {episodesQuery} from "../../../client/Queries";
import ErrorHandler from "../../../components/LoginSystem/guards/errorHandler";

const Episodes = () => {

  const { data, loading, error } = useQuery(episodesQuery);

  if (loading) return 'Loading ...';
  if (error)return (<ErrorHandler/>);

  const [...allEpisodes] = data.allEpisodes.edges;

  return (
          <Row>
              {
                allEpisodes.map(({ node }) => (
                        <Col md={4} key={node.episodeId}>
                          <Link className={ styles.cardPanel }
                                as={RouterLink}
                                variant="nav"
                                key={node.episodeId}
                                to={`/episodes/${node.episodeId}`}>
                            <Card variant="primary" className={`${styles.cardPanel}`}>
                              <CardImg top width="100%" src={node.image} className={`${styles.thumbnail} img-fluid`} alt="Episode" />
                              <div className={`${styles.cardBody}`} >
                                <Heading variant="primary" className={`${styles.title} title-global`}>{node.title}</Heading>
                                <CardText className={`${styles.text}`} >{node.openingCrawl.substring(0, 221) + "..."}</CardText>
                              </div>
                            </Card>
                          </Link>
                        </Col>
                       ))
              }
          </Row>);
};

export default Episodes;

