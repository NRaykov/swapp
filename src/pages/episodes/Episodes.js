import React from 'react';
import Login from '../../components/Login/Login.js';
import { isAuthenticated, signout } from '../../guards/auth.js';
import { withRouter } from 'react-router-dom';

import { Row, Col, Card, CardImg, CardBody, CardTitle, CardText } from 'reactstrap';
import styles from './styles.module.css';


//Placeholder
const episodes = [0,1,2,3,4,5];

const Episodes = withRouter(({ history }) =>
            isAuthenticated ? (
                    <React.Fragment>
                      <Row>
                        {/*TODO Fetch Data from DB*/}
                        { episodes.map((index) => {
                          return (
                            <Col md="4" key={index}>
                              <Card className={styles.cardPanel}>
                                <CardImg top width="100%" src="https://via.placeholder.com/400x400" className="img-fluid" alt="Card image cap" />
                                <CardBody>
                                  <CardTitle className={styles.title}>Card title</CardTitle>
                                  <CardText>Some quick example text to build on the card title and make up the bulk of the card's content.</CardText>
                                </CardBody>
                              </Card>
                            </Col>
                          )
                        }) }

                      </Row>
                    </React.Fragment>

            ) : ( <Login/> ),
        );
export default Episodes;
