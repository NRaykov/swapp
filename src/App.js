import React, {useState} from 'react';
//Apollo Boost
import { ApolloProvider } from '@apollo/react-hooks';
import ApolloClient from "apollo-client"

import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';

import authLink from './client/auth';
import {typeDefs} from './client/local'

//Navigation and Login
import { BrowserRouter } from 'react-router-dom';
import Container from './components/StylesComponents/elements/container/container';
import Header from "./components/Header/Header";
import Home from "./pages/HomeComponent/Home";
//Styling
import './App.css';
import { ThemeProvider } from 'styled-components';
import ThemeChanger from './components/StylesComponents/themeChanger';
import { themes } from './components/StylesComponents';

    const httpLink = createHttpLink({
      uri: 'https://swapp.st6.io/graphql',
    });

    const cache = new InMemoryCache();

    //Export 'client' to use it outside of this scope
    export const client = new ApolloClient({
      cache,
      link: authLink.concat(httpLink),
      resolvers: {
          /** TODO  Create Resolvers Here **/
      },
      typeDefs,
    });

    const isAuthed = !!localStorage.getItem('token');
    cache.writeData({
      data: {
        authenticated: isAuthed,
      },
    });


export const AppComponent = () => {

    /**** Set Theme to LocalStorage ****/
    const [theme, setTheme] = useState('light');
    const themeChanger = ThemeChanger(theme, setTheme);

    localStorage.setItem('Theme', 'light');
    const getTheme = localStorage.getItem('Theme');
    console.log(getTheme);


  return (
        <ApolloProvider client={client}>
            <BrowserRouter>
              <ThemeProvider theme={themes[getTheme]}>
                <Container variant="primary" className="container-main">
                    <Header themeChanger={themeChanger}/>
                    <div className="container py-5">
                      <Home themeChanger={themeChanger}/>
                    </div>
                </Container>
              </ThemeProvider>
            </BrowserRouter>
        </ApolloProvider>
  )
};

