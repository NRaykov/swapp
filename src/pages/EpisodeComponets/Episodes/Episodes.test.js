import React from 'react';
import Episodes from './Episodes'
import { ApolloProvider } from "react-apollo";
import '../../../setupTests';
import { client } from '../../../App'
const ReactTestRenderer = require('react-test-renderer');

describe('Snapshot test "Episodes" component', () => {

  it('Should compare the component with a snapshot', () => {
    const component = ReactTestRenderer.create(
            <ApolloProvider client={client}>
              <Episodes />
            </ApolloProvider>
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })
});
