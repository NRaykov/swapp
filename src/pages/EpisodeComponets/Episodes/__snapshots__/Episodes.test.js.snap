// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test "Episodes" component Should compare the component with a snapshot 1`] = `"Loading ..."`;
